src/components/AvatarStack/__stories__/AvatarStack.stories.tsx

```diff
@@ -0,0 +1,108 @@
+
+import {faker} from '@faker-js/faker/locale/en';
+import type {Meta, StoryFn} from '@storybook/react';
+
+import type {AvatarProps, AvatarSize} from '../../Avatar';
+import {AVATAR_SIZES, Avatar, DEFAULT_AVATAR_SIZE} from '../../Avatar';
+import {Tooltip} from '../../Tooltip';
+import {AvatarStack} from '../AvatarStack';
+
+type ComponentType = React.ComponentProps<typeof AvatarStack> & {avatarSize: AvatarSize};
+
+function getChildren({
+    count = faker.number.int({min: 1, max: 30}),
+    avatarSize = DEFAULT_AVATAR_SIZE,
+}: Partial<{count: number; avatarSize: AvatarSize}>) {
+    return faker.helpers.uniqueArray(
+        () => (
+            <Avatar
+                imgUrl={faker.image.avatar()}
+                size={avatarSize}
+                borderColor={'var(--g-color-line-generic-solid)'}
+                aria-label={'For tests'}
+                alt={'For tests'}
+            />
+        ),
+        count,
+    );
+}
+
+export default {
+    title: 'Components/Data Display/AvatarStack',
+    component: AvatarStack,
+    args: {
+        overlapSize: 's',
+        avatarSize: DEFAULT_AVATAR_SIZE,
+    },
+    argTypes: {
+        avatarSize: {
+            control: 'select',
+            options: Object.keys(AVATAR_SIZES),
+            name: 'Size of avatar',
+            description: 'Not part of component API',
+        },
+    },
+} as Meta<ComponentType>;
+
+function getTemplateChildren(args: ComponentType) {
+    const {avatarSize} = args;
+
+    return (
+        React.Children.map(args.children, (child) =>
+            React.isValidElement<AvatarProps>(child)
+                ? React.cloneElement(child, {
+                      size: avatarSize,
+                      borderColor: 'var(--g-color-line-generic-solid)',
+                  })
+                : null,
+        ) || getChildren({avatarSize})
+    );
+}
+
+const Template: StoryFn<ComponentType> = (args) => {
+    const {avatarSize: _, ...props} = args;
+    return <AvatarStack {...props}>{getTemplateChildren(args)}</AvatarStack>;
+};
+
+const TemplateWithButtonOverride: StoryFn<ComponentType> = (args) => {
+    const {avatarSize, ...props} = args;
+
+    return (
+        <AvatarStack {...props}>
+            {getTemplateChildren(args)}
+            <AvatarStack.MoreButton
+                key="more-button-override"
+                size={avatarSize}
+                aria-label={'Rest of the users'}
+                count={24}
+                render={({button}) => (
+                    <Tooltip content={'Somehow display list of all other items'}>{button}</Tooltip>
+                )}
+            />
+        </AvatarStack>
+    );
+};
+
+export const Default = Template.bind({});
+
+export const WithOneItem = Template.bind({});
+WithOneItem.args = {
+    children: getChildren({count: 1}),
+};
+
+export const WithMoreButton = Template.bind({});
+WithMoreButton.args = {
+    children: getChildren({count: 6}),
+};
+
+export const EdgeCase = Template.bind({});
```

Comment:
Name isn't descriptive enough. What this story is showing?

Reply:
Edge case when only 1 item hidden, in that case we didn't render more button, and just displays item

Reply:
Why we didn't render `+1` button?